lib/circuit.js

```diff
@@ -241,6 +253,25 @@ class CircuitBreaker extends EventEmitter {
     if (this.options.cache) {
       CACHE.set(this, undefined);
     }
+
+     // Prepopulate the State of the Breaker
+    if (this[CLOSED]) {
+      // If the state is closed, we don't do anything?
+      this[STATE] = CLOSED;
+      this.close();
+    } else if (this[OPEN]) {
+      // If the state being passed in is OPEN
+      // THen we need to start some timers
+      this.open();
+    } else if (this[HALF_OPEN]) {
+      // Not sure if anythign needs to be done here
+      this[STATE] = HALF_OPEN;
+    } else if (this[SHUTDOWN]) {
+      // IF the state of the circuit being passed in is SHUTDOWN
+      // Then i don't think we need to do anything either?
+      this[STATE] = SHUTDOWN;
+      this.shutdown();
+    }
```

Comment:
This should all be in the above `if (options.state) {}` block, shouldn't it? And could probably be simplified to something like this.

```suggestion
     if (options.state.shutdown) {
       this.shutdown();
     } else if (options.state.open) {
       this.open(); 
     }

     if (options.state.halfOpen) {
       this[STATE] = HALF_OPEN;
     }
```

Reply:
I'm pretty sure there was a reason for doing this way.  I think it was becuase there are event listeners and functions that need to get created first,  before this code gets executed.  but i'll have to go back and check that